cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt

```diff
@@ -181,43 +181,43 @@ class ExpressionHandler(frontend: PythonLanguageFrontend) :
         return subscriptExpression
     }
 
+    /**
+     * Handles a [Python.AST.BoolOp]. This can be either a chain of `and` or `or` expressions. We
+     * model this as [BinaryOperator]s with nesting. Thus, `a and b and c` becomes `(a and b) and c`
+     * (same for `or`).
+     */
     private fun handleBoolOp(node: Python.AST.BoolOp): Expression {
         val op =
             when (node.op) {
                 is Python.AST.And -> "and"
                 is Python.AST.Or -> "or"
             }
-        return handleBoolOpRecursively(node.values, op, node)
-    }
+        return when (node.values.size) {
+            0,
+            1 ->
+                newProblemExpression(
+                    "handleBoolOp: Expected at least two expressions but got " + node.values.size,
+                    rawNode = node
+                )
+            else -> {
+                val result = newBinaryOperator(operatorCode = op, rawNode = node)
+                var idx = 0
+                var currentInnerNode: BinaryOperator = result
+
+                // For all iterations until the last one: set the lhs to handle the current idx and
+                // create a new nested [BinaryOperator] for the rhs. Then slide the window by one.
+                while (idx < node.values.size - 2) {
+                    currentInnerNode.lhs = handle(node.values[idx])
+                    currentInnerNode.rhs = newBinaryOperator(operatorCode = op, rawNode = node)
+                    currentInnerNode = currentInnerNode.rhs as BinaryOperator
```

Comment:
Any Kotlin magic to get rid of this unchecked cast?

Reply:
val binOp = newBinaryOperator(operatorCode = op, rawNode = node)
currentInnerNode.rhs = binOp 
currentInnerNode = binOp

Reply:
implemented